src/sciline/scheduler.py

```diff
@@ -107,3 +112,13 @@ def __repr__(self) -> str:
         module = getattr(inspect.getmodule(self._dask_get), '__name__', '')
         name = self._dask_get.__name__
         return f'{self.__class__.__name__}({module}.{name})'
+
+
+def _to_dask_key(key: Hashable) -> str:
+    """Map a Sciline key to a dask key.
+
+    According to the docs (https://docs.dask.org/en/stable/spec.html#definitions),
+    keys in a Dask graph are not allowed to be types.
+    So this function converts Sciline keys (types) to strings.
+    """
+    return key_full_qualname(key)
```

Comment:
I wonder if we should do this in a base scheduler? But probably does not matter now, we can change later if ever needed elsewhere but dask?

... or does this matter for JSON serialization? Is there some duplication to be avoided if this moves to a lower level? Or to `TaskGraph`?

Reply:
As far as scheduling is concerned, the conversion could be done in `TaskGraph`. But it is lossy because it looses the actual domain type. So it may get in the way of other operations than scheduling.

It could be done in a base scheduler if you want it to create a separate intermediate graph with replaced keys. But given that it is only an issue for dask at the moment and that we don't have plans for any other scheduler (as far as I know), I would not generalise yet. We can always do that later if need be.

Reply:
What about the serialization and deserialization part? Was there any trouble with the latter, reconstructing the key types?

Reply:
It serializes the dask graph which only contains strings as keys. So no, this part does not seem to cause problems. Only providers are problematic.